### PR TITLE
Add google pub/sub doc

### LINKED ISSUE #1474: document: google pubsub source
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

Although barely requested, we still maintain the code of pubsub source. Thus, we need documentation as well.

### Reference

_No response_

### LINKED ISSUE #1775: Document: feat: pre-release pubsub
This issue tracks the documentation update needed for the merged PR #14531.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/14531
Source PR Merged At: 2024-01-16T07:45:59Z

### PR DESCRIPTION
**Pull request by WanYixian**
## Info

- **Description**

     Add google pub/sub doc

- **Related code PR**

   https://github.com/risingwavelabs/risingwave/pull/14531

- **Related doc issue**
  
  Resolves 
  https://github.com/risingwavelabs/risingwave-docs/issues/1474
  https://github.com/risingwavelabs/risingwave-docs/issues/1775

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/ingest/ingest-from-google-pubsub.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+   schema.location = 'location' | schema.registry = 'schema_registry_url'
+);
+```
+
+## Parameters
+
+| Field | Note |
+| --- | --- |
+| pubsub.subscription | Required. Specifies the Pub/Sub subscription to consume messages from. Ensure the subscription is configured with the `retain-on-ack` property to enable message replay. |
+| pubsub.emulator_host | Optional. Enables the connector to work with a Pub/Sub emulator. Do not configure in the production. |
```
Comment (tabVersion) [inline docs/ingest/ingest-from-google-pubsub.md]: the param is only used for local test. can remove in user-facing doc

File: docs/ingest/ingest-from-google-pubsub.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+| pubsub.start_offset.nanos | Optional. Cannot be set together with pubsub.start_snapshot. Specifies a numeric timestamp in nanoseconds, ideally the publish timestamp of a message in the subscription. If present, the connector seeks the subscription to the timestamp and starts consuming from there. Note that the seek operation is subject to limitations based on the message retention policy of the subscription.|
+| pubsub.start_snapshot | Optional. Cannot be set together with `pubsub.start_offset.nanos`. If present, the connector first seeks to the specified snapshot before starting consumption. |
+|  |  |
+
+:::note
+The Pub/Sub topic provided must have `retain_acked_messages` enabled and must define a retention policy. For details, see [Configure subscription message retention](https://cloud.google.com/pubsub/docs/replay-overview#subscription_message_retention).
+:::
+
+We can only achieve at-least-once semantic for the Pub/Sub source rather than exactly once because the SDK cannot seek back to a specific message offset. Here's how the recovery process works:
```
Comment (tabVersion) [inline docs/ingest/ingest-from-google-pubsub.md]: we should highlight this part because we offer weaker semantic guarantee on this connector.

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: docs/ingest/ingest-from-google-pubsub.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   schema.location = 'location' | schema.registry = 'schema_registry_url'
+);
+```
+
+## Parameters
+
+| Field | Note |
+| --- | --- |
+| pubsub.subscription | Required. Specifies the Pub/Sub subscription to consume messages from. Ensure the subscription is configured with the `retain-on-ack` property to enable message replay. |
+| pubsub.credentials | Required. a JSON string containing the service account credentials for authorization, see the [service-account credentials guide](https://developers.google.com/workspace/guides/create-credentials#create_credentials_for_a_service_account). The provided account credential must have the `pubsub.subscriber` [role](https://cloud.google.com/pubsub/docs/access-control#roles). |
```
Comment (ShanlanLi) [inline docs/ingest/ingest-from-google-pubsub.md]:
```suggestion
| pubsub.credentials | Required. A JSON string containing the service account credentials for authorization, see the [service-account credentials guide](https://developers.google.com/workspace/guides/create-credentials#create_credentials_for_a_service_account). The provided account credential must have the `pubsub.subscriber` [role](https://cloud.google.com/pubsub/docs/access-control#roles). |
```

File: docs/ingest/ingest-from-google-pubsub.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+| pubsub.start_offset.nanos | Optional. Cannot be set together with pubsub.start_snapshot. Specifies a numeric timestamp in nanoseconds, ideally the publish timestamp of a message in the subscription. If present, the connector seeks the subscription to the timestamp and starts consuming from there. Note that the seek operation is subject to limitations based on the message retention policy of the subscription.|
+| pubsub.start_snapshot | Optional. Cannot be set together with `pubsub.start_offset.nanos`. If present, the connector first seeks to the specified snapshot before starting consumption. |
+|  |  |
```
Comment (ShanlanLi) [inline docs/ingest/ingest-from-google-pubsub.md]:
```suggestion
```

File: docs/ingest/ingest-from-google-pubsub.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+Here's how the recovery process works:
+
+1. Record the timestamp for each message as an offset.
+
+2. During recovery, seek back to the specific timestamp and consume messages again from the topic.
+
+If two messages are produced almost at the same time, it is possible to receive each message more than once.
+
+## Example
```
Comment (ShanlanLi) [inline docs/ingest/ingest-from-google-pubsub.md]:
```suggestion
## Examples
```

**Review by ShanlanLi — APPROVED** [review]
Rest LGTM

**Pull request merged**